### PR TITLE
Add children cards to choral view page

### PR DESCRIPTION
**Pull request by jboretsky**
Haven't really styled this yet other than positioning, but we can make the app look nice today.

Sorry about the very long promise chain, I tried to split it up into two parts: children stuff, and parent stuff. Once all of that is done, it renders the page.

### CODE REVIEW

**Comment by jboretsky** [comment]
image

**Comment by andreirtaylor** [comment]
This is glorious.

**Comment by jboretsky** [comment]
Thanks! :blush:

File: views/choral.pug [hunk 1]
```diff
@@ -63,16 +78,32 @@ block content
...
           }
         }
       });
       socket.emit('subscribeToID', { choralId: '#{parentChoralId}' });
+      for(var id in childData) {
+        if(childData.hasOwnProperty(id)) {
+          socket.emit('subscribeToID', { choralId: id });
+        }
+      }
+      console.log(childData);
```
Comment (james-woo) [inline views/choral.pug]: Maybe remove these console.logs

File: views/choral.pug [hunk 2]
```diff
@@ -63,16 +78,32 @@ block content
...
       socket.on('ackConnect', function (data) {
```
Comment (james-woo) [inline views/choral.pug]: Does this actually do anything?

Comment (jboretsky) [reply views/choral.pug]: which part? I think 'ackConnect' isn't being used anymore so I could remove it

File: views/choral.pug [hunk 3]
```diff
@@ -18,8 +18,23 @@ block content
...
+            - for( var entry in device_data )
+              - var data_value = device_data[entry]
+              - if( !isNaN(parseFloat(data_value)) && isFinite(data_value) )
+                h5.child-card-body-label= entry
+                h5.child-card-body-data(id=child.choralId class=entry)= JSON.stringify(device_data[entry])
+
   script.
     document.addEventListener("DOMContentLoaded", function(event) {
+      var childData = !{JSON.stringify(childData)}
+      console.log(childData);
```
Comment (james-woo) [inline views/choral.pug]: Remove console.log

**Comment by cpjk** [comment]
Those callbacks
thousand_yard_stare_dog

**Comment by cpjk** [comment]
lgtm

**Pull request merged**